### ISSUE TITLE
Confusion between parenthesis and generator if inside a list comprehension

### ISSUE THREAD

**Issue by guillaume-alliander** [microsoft/pyright#7481]
**Describe the bug**
If brackets are (by mistake) around the inside of a list comprehension, pyright does not seem to notice a generator is built. 

**Code or Screenshots**

Buggy code: pyright is happy, running fails (as it should, but because pyright was happy, I didn't expect it).
```python
from typing import reveal_type

l1 = [(i for i in [1, 2, 3])]
reveal_type(l1[0])  # says int during pyright run, generator during python run
l2 = [i * 2 for i in l1] # BOOM unsupported operand type(s) for *: 'generator' and 'int'
```

Rewriting the same code with one extra manual step, pyright is unhappy, as it should.
```python
from typing import reveal_type

g = (i for i in [1, 2, 3])
l1 = [g]
reveal_type(l1[0])  # both pyright and python say it's a generator
l2 = [i * 2 for i in l1] # pyright does complain here (and the python run obviously fails as well)
```

**VS Code extension or command-line**
Both. Second version is found inside Vscode, first version is not found on command line or inside vscode. 
> pyright --version
> pyright 1.1.354

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.355, which I just published.
